### PR TITLE
#11108 Disabled hotkeys in scenarios where their usage was not intended

### PR DESCRIPTION
**Pull request by IlijaQ**
Fixes [#11108](https://github.com/gitextensions/gitextensions/issues/11108)

## Proposed changes

- Creating a new branch using hotkeys `Ctrl + B` does not prompt `FromCreateBranch` while in bare repository, neither when artificial commit is selected.
- While selecting multiple commits, hotkey for creating new branch is disabled
- Rebasing current branch using hotkeys `Ctrl + Shift + E` is disabled while artificial commit is selected
- Creating a new tag using hotkeys `Ctrl + T` is disabled while artificial commit is selected

## Test methodology

- Manual with GE repo

## Test environment(s)

- Git Extensions 33.33.33
- Build https://github.com/gitextensions/gitextensions/commit/8b5eab531dcc849d00110df6f9797a096e68b500
- Git 2.30.0.windows.2 (recommended: 2.40.1 or later)
- Microsoft Windows NT 10.0.19045.0
- .NET 6.0.14
- DPI 120dpi (125% scaling)

### CODE REVIEW

**Review by gerhardol — COMMENTED** [review]
Most similar issues should be handled by the check in the method returns false when conditions are not fulfilled (the action could be handled by another form then). This should be done here too.
I will not oppose a merge though, if another maintainer approves.

I am sure there are several other similar issues still even if several similar issues exists...

**Comment by IlijaQ** [comment]
Yes, I have tried that. When the method returns false, the hotkey is passed to the context menu. Every time we open the context menu for the selected commit, it checks if create branch action or some other action should be enabled, but that is to late.

The problem is when we select a regular commit, right click, and then select an artificial commit - without right clicking on the artificial commit, and press the hotkey, we can still cause this bug.

If you are still in favor of passing the action handling to another form, we may want to add a double redundant validation in context menu click event handeling or change the context menu to perform update of Enabled propery of all context menu items every time a new commit is selected.

**Comment by gerhardol** [comment]
I just want to move the logic ti the method handling the request, not in `ExecuteCommand()`.
This is how all other hot keys are handled.

**Comment by IlijaQ** [comment]
Ok,
Just in case, please specify the method where we want to move the validations to. Do we want to move the validations to the corresponding method in the switch statement? e.g. in case Command.CreateBranch: should we move the validation to StartCreateBranchDialog?

**Comment by IlijaQ** [comment]
@gerhardol I have updated the code for your review, according to your suggestions.

File: GitUI/GitUICommands.cs
Comment (gerhardol) [inline GitUI/GitUICommands.cs:443]:
```suggestion
            if (Module.IsBareRepository() || objectId?.IsArtificial is true)
```

File: GitUI/GitUICommands.cs
Comment (gerhardol) [inline GitUI/GitUICommands.cs:861]:
```suggestion
            if (objectId?.IsArtificial is true)
```

File: GitUI/CommandsDialogs/FormBrowse.cs
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormBrowse.cs:1528]: Note for other reviewers: This check is not in GitUICommands, as the string is passed, not the ObjectId so each caller must check.

**Review by gerhardol — APPROVED** [review]
Approved with two suggested changes.
Changing from draft.

**Review by gerhardol — APPROVED** [review]

**Comment by IlijaQ** [comment]
Sorry, my bad. I missed the question marks in my last commit. It should pass the build now.
It needs one more approval.
I apologize.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you!

**Pull request merged**
